LocalSettings.php

```diff
@@ -201,199 +213,344 @@
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-view'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-log'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-log-detail'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-modify-restricted'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-revert'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-view-private'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-log-private'] = true;
-$wgRevokePermissions['abusefilter-restricted']['abusefilter-modify-blocked-external-domains'] = true;
-$wgAddGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 
-'researcher', 'chatmod', 'blockedfromchat'  );
-$wgRemoveGroups['sysop'] = array( 'editor', 'reviewer', 'autoreview', 'autopatrol', 'patroller', 'translateadmin', 'confirmed', 
-'researcher', 'chatmod', 'blockedfromchat' );
-$wgAddGroups['steward'] = array( 'sysop', 'interface-admin', 'bureaucrat', 'steward', 'checkuser', 'suppress', 'non-stewardsuppress', 'abusefilter-admin', 'abusefilter-restricted' );
-$wgRemoveGroups['steward'] = array( 'sysop', 'interface-admin', 'bureaucrat', 'steward', 'checkuser', 'suppress', 'non-stewardsuppress', 'abusefilter-admin', 'abusefilter-restricted' );
-$wgAddGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat', 'interwiki-admin' );
-$wgRemoveGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat', 'interwiki-admin' );
-$wgGroupsRemoveFromSelf['sysop'] = array( 'sysop' );
-$wgRestrictionLevels = array( '', 'user', 'bureaucrat', 'sysop', 'autoconfirmed', 'steward' );
-$wgRestrictionTypes = array( 'create', 'edit', 'move', 'upload', 'delete', 'protect' );
-$wgCascadingRestrictionLevels = array( 'bureaucrat', 'sysop', 'steward' );
-$wgSemiprotectedRestrictionLevels = array( 'user', 'autoconfirmed', );
+
+// Permission assignments
+/// Administrators
+$wgAddGroups['sysop'] = array(
+	'autopatrol',
+	'autoreview',
+	'blockedfromchat',
+	'chatmod',
+	'confirmed',
+	'editor',
+	'patroller',
+	'researcher',
+	'reviewer',
+	'translateadmin'
+);
+$wgRemoveGroups['sysop'] = array(
+	'autopatrol',
+	'autoreview',
+	'blockedfromchat',
+	'chatmod',
+	'confirmed',
+	'editor',
+	'patroller',
+	'researcher',
+	'reviewer',
+	'translateadmin'
+);
+
+/// Bureaucrats
+$wgAddGroups['bureaucrat'] = array(
+	'autoconfirmed',
+	'autopatrol',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+$wgRemoveGroups['bureaucrat'] = array(
+	'autoconfirmed',
+	'autopatrol',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+
+/// Stewards
+$wgAddGroups['steward'] = array(
+	'autoconfirmed',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'steward',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+$wgRemoveGroups['steward'] = array(
```

Comment:
stewards are currently able to remove steward, that shouldn't change.

Reply:
See 997e304